test/unit/alphabet/composition/union_composition_test.cpp

```diff
@@ -311,3 +311,46 @@ TEST(union_composition_test, value_size)
     EXPECT_EQ(alphabet2_t::value_size, 10);
     EXPECT_EQ(alphabet3_t::value_size, 1);
 }
+
+TEST(union_composition_test, convert_by_index)
+{
+    union_composition<dna4, dna5, gap> u;
+    u = dna5::C;
+
+    EXPECT_FALSE(u.is_alternative<0>());
+    EXPECT_TRUE(u.is_alternative<1>());
+    EXPECT_FALSE(u.is_alternative<2>());
+
+    EXPECT_THROW(u.convert_to<0>(), std::bad_variant_access);
+    EXPECT_NO_THROW(u.convert_to<1>());
+    EXPECT_THROW(u.convert_to<2>(), std::bad_variant_access);
+
+    dna5 out = u.convert_to<1>();
+    EXPECT_EQ(out, dna5::C);
+
+    u = gap::GAP;
+
+    gap g = u.convert_unsafely_to<2>();
+    EXPECT_EQ(g, gap::GAP);
+}
+
+TEST(union_composition_test, convert_by_type)
+{
+    union_composition<dna4, dna5, gap> u;
```

Comment:
maybe add a test for a `union_composition<dna4, dna4> letter{std::in_place_index_t<1>{}, dna4::A};`
which is converted to dna4